src/mock/parsec/state.rs

```diff
@@ -55,13 +58,19 @@ impl<T: NetworkEvent, P: PublicId> SectionState<T, P> {
     }
 
     pub fn compute_consensus(&mut self, peers: &BTreeSet<P>, consensus_mode: ConsensusMode) {
-        for holder in mem::replace(&mut self.unconsensused_observations, BTreeSet::new()) {
-            let state = unwrap!(self.observations.get_mut(&holder));
-            if let Some(block) = state.compute_consensus(peers, consensus_mode, &holder) {
-                self.blocks.push((block, holder));
-            } else {
-                let _ = self.unconsensused_observations.insert(holder);
+        let mut unconsensused_genesis = false;
+
+        for holder in mem::replace(&mut self.unconsensused_observations, Vec::new()) {
+            if !unconsensused_genesis {
```

Comment:
This block could use a comment - I'm not sure what the idea is here.

Reply:
will add comment:
```
// Update consensus and add block for now consensused observations, keep remaining unconsensused_observations.
// If the genesis is unconsensused, keep all observation as unconsensused, as genesis needs to be the first consensused block.
```

Reply:
clarified and added:
        // Update consensus for unconsensused_observations.
        // Add blocks for consensused observations and keep others in unconsensused_observations.
        // Genesis needs to be the first block: If it is unconsensused, no block is added.

Reply:
I still don't get it, to be honest.
`unconsensused_genesis` is initialised to false. If the first consensused block _isn't_ genesis, it will actually get added to `self.blocks`. If genesis appears somewhere later, it will cause all the blocks _after it_ to be left in `unconsensused_observations`. So as far as I can understand, this will make all blocks _up until genesis_ consensused... or am I missing something?

Reply:
Genesis will not appear sometime later because Genesis is voted first when the Parsec is created.

Reply:
That makes sense, but why do we need these precautions, then? Wouldn't it get consensused as the first block, anyway?

Reply:
Well the test was failing :-) so yes we need to take this precaution:
Genesis is supermajority vote and so within the mock implementation will get consensus once enough new parsec instance have been created, Opaque observations are single majority vote, and so will consensus immediately, as a result, if you vote for Opaque immediatly after creation, that opaque comes out first.

Reply:
I see :) Could you mention this in the comment, too? I think it would help a lot.

Reply:
I did put some info in the commit comment.
Which specific part do you want in the code comment?

Reply:
Maybe that if the Genesis is among unconsensused observations, it will be the first one, but it might still not get consensused as the first block, because it requires a supermajority, and later blocks might only require a single vote. That's the part that confused me, at least.